tests/test_series.py

```diff
@@ -1165,14 +1213,20 @@ def test_series_dtype() -> None:
 def test_types_replace() -> None:
     # GH 44
     s = pd.Series([1, 2, 3])
-    check(assert_type(s.replace(1, 2), pd.Series), pd.Series)
-    check(assert_type(s.replace(1, 2, inplace=False), pd.Series), pd.Series)
+    check(assert_type(s.replace(1, 2), "pd.Series[int]"), pd.Series, np.intp)
+    check(
+        assert_type(s.replace(1, 2, inplace=False), "pd.Series[int]"),
+        pd.Series,
+        np.intp,
+    )
     assert assert_type(s.replace(1, 2, inplace=True), None) is None
 
 
 def test_cat_accessor() -> None:
     # GH 43
-    s = pd.Series(pd.Categorical(["a", "b", "a"], categories=["a", "b"]))
+    s: pd.Series[str] = pd.Series(
+        pd.Categorical(["a", "b", "a"], categories=["a", "b"])
```

Comment:
Maybe we should have a `CategoricalSeries` ?

Reply:
I think we might need to reconsider S1 at some point (not in this PR). I would slightly prefer `S1 = TypeVar("S1")` as any type can be in a Series.

Reply:
Yes, but I'd rather encourage "typical" types to be in a `Series`, and let people do `cast` or `ignore` if they are putting other types in a `Series`.

Reply:
We could do a `CategoricalSeries` as a future PR